.github/workflows/lint.yml

```diff
@@ -5,11 +5,11 @@ on: push
 jobs:
   black:
 
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
     
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v4
       - uses: psf/black@stable
```

Comment:
I've found it better to install Black (with a specific version) as a dev dependency on and invoke it using Python in the CI and from the command line or IDE when developing. Otherwise different versions of Black disagree on their defaults (`@stable` does change), and it is very frustrating. Further, different repos following this patter use different versions, so it is not possible at present to have one global version on your machine.

Reply:
Can you point to a similar config in another repo that we could adopt in this one? (I'm not finding a lint.yml in any other repositories).

My approach to using `black` has essentially been asking it to fix our formatting for us, and its version hasn't been of any consequence.

Reply:
It is possible that Black has settled down lately, but in the past it changed its default settings from release to release. Which caused angst when it changed under you -- potentially hundreds of changes in a formerly Black-compliant codebase. Hence version pinning.

I've noticed that Black checks are often the last step of the Python CI workflow. For example, [in pycds](https://github.com/pacificclimate/pycds/blob/master/.github/workflows/python-ci.yml). I believe this is because a failing Black check can cause your unit tests to be cancelled, and this can be undesirable. I think I recall discussing this with Nik a while back when we were both working on the same repo. In any case, it is common practice.

Reply:
I'd argue that it's a good thing for black checks to be done *first* since they are the cheapest and that we consider them to be important for controlling code quality.

If there is a specific version of black that we should pin to, please feel free to recommend it :)